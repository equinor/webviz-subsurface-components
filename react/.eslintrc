{
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module"
  },
  "extends":  [
    "plugin:react/recommended",
    "eslint:recommended",
    "plugin:prettier/recommended"
  ],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx"],
      "plugins": ["@typescript-eslint"],
      "extends": [
        "plugin:react/recommended",
        "plugin:@typescript-eslint/recommended",
        "plugin:prettier/recommended"
      ]
    }
  ],
  "settings": {
    "react": {
      "version": "detect"
    }
  },
  "env": {
    "amd": true,
    "browser": true,
    "es6": true,
    "jasmine": true,
    "node": true
  }
}
