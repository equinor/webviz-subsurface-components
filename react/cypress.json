{
    "baseUrl": "http://localhost:6006/?path=/story/deckglmap--default",
    "video": true
}
