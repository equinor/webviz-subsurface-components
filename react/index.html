<html>
    <head>
        <title>Demo application of webviz-subsurface-components</title>
    </head>
    <body>
        <div id='root'></div>
        <link rel="stylesheet" type="text/css" href="output.css">
        <script src="output.js"></script>
    </body>
</html>
