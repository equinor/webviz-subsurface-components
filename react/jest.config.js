/**
 * The jest tests had an error: SyntaxError: Cannot use import statement outside a module
 * After some investigation, it leads to this page https://github.com/facebook/jest/issues/9395.
 */
module.exports = {
    moduleNameMapper: {
        "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$":
            "<rootDir>/__mocks__/fileMock.js",
        "\\.(css|less)$": "<rootDir>/__mocks__/styleMock.js",
    },
    transform: {
        "\\.(js|ts|jsx|tsx)$": [
            "babel-jest",
            { configFile: "./config/babel.config.json" },
        ],
    },
    transformIgnorePatterns: ["<rootDir>/node_modules/(?!(@webviz)/)"],
};
