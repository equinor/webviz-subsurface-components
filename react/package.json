{
    "name": "@webviz/subsurface-components",
    "description": "Custom Dash components for use in Webviz",
    "repository": {
        "type": "git",
        "url": "git://github.com/equinor/webviz-subsurface-components.git"
    },
    "bugs": {
        "url": "https://github.com/equinor/webviz-subsurface-components/issues"
    },
    "homepage": "https://github.com/equinor/webviz-subsurface-components",
    "main": "dist/index.js",
    "types": "dist/index.d.ts",
    "files": [
        "dist/**/*"
    ],
    "scripts": {
        "start": "webpack serve --mode development --devtool inline-source-map --entry ./src/demo/index.js --open",
        "copy-package-json": "copyfiles ./package.json ../webviz_subsurface_components/",
        "transpile": "tsc --project tsconfig.build.json && npm run copy-files",
        "transpile:dash": "tsc --project tsconfig.build.json && rimraf ./dist/components/*/**/",
        "tidy-up": "rimraf ./dist ./webviz_subsurface_components",
        "build:js": "NODE_ENV=production webpack --mode production",
        "build:js-dev": "NODE_ENV=development webpack --mode development",
        "build:js-demo": "NODE_ENV=production webpack --mode production --entry ./src/demo/index.js",
        "build:py": "npm run transpile:dash && copyfiles -a ../webviz_subsurface_components/. ./webviz_subsurface_components/ && copyfiles ./package.json ./webviz_subsurface_components/ && copyfiles -f ./src/lib/assets/VectorDefinitions.json ./webviz_subsurface_components/ && dash-generate-components ./dist/components webviz_subsurface_components -p package.json --ignore '(^index.js|.stories.js)$' && copyfiles -a -f ./webviz_subsurface_components/* ../webviz_subsurface_components/",
        "build": "npm run transpile && npm run build:js && npm run build:js-dev && npm run tidy-up && npm run build:py && npm run tidy-up",
        "typecheck": "tsc --noEmit",
        "format": "eslint --fix *.js *json \"src/**/*.+(ts|tsx|js|jsx|json|css)\"",
        "lint": "eslint *.js *.json \"src/**/*.+(ts|tsx|js|jsx|json|css)\"",
        "validate": "npm run typecheck && npm run lint",
        "setup-deckgl-types": "cd ./node_modules/@danmarshall/deckgl-typings/ && indefinitely-typed --folder deck.gl__aggregation-layers --folder deck.gl__core --folder deck.gl__extensions --folder deck.gl__geo-layers --folder deck.gl__google-maps --folder deck.gl__json --folder deck.gl__layers --folder deck.gl__mapbox --folder deck.gl__mesh-layers --folder deck.gl__react --folder deck.gl --folder luma.gl__constants --folder luma.gl__core --folder luma.gl__gltools --folder luma.gl__webgl-state-tracker --folder luma.gl__webgl --folder math.gl__core --folder math.gl",
        "storybook": "start-storybook -p 6006",
        "build-storybook": "build-storybook",
        "test_perf": "jest _performance",
        "test_correctness": "jest --coverage --testPathIgnorePatterns='_performance'",
        "test": "jest --coverage",
        "test:watch": "npm test -- --watch",
        "build:npm": "rimraf dist && tsc -project tsconfig.build.json --declaration && npm run copy-files",
        "copy-files": "copyfiles package.json dist && copyfiles --up 1 ../README.md dist && copyfiles --exclude **/*.{tsx,jsx,js,ts,snap} --up 2 \"src/lib/**/*\" dist/",
        "prepare": "npm run build:npm",
        "postpack": "npm run tidy-up",
        "prepublishOnly": "npm test && npm run lint",
        "preversion": "npm run lint",
        "test-cypress": "concurrently 'npm run storybook' 'npm run cypress:test' -k -s first",
        "cypress:test": "wait-on http://localhost:6006 && cypress run",
        "cypress:run": "cypress run"
    },
    "author": "Equinor <opensource@equinor.com>",
    "license": "MPL",
    "dependencies": {
        "@emerson-eps/color-tables": "^0.4.22",
        "@equinor/eds-core-react": "^0.12.1",
        "@equinor/eds-icons": "^0.9.1",
        "@equinor/videx-wellog": "^0.7.0",
        "@material-ui/core": "^4.12.3",
        "@material-ui/lab": "^4.0.0-alpha.60",
        "@react-leaflet/core": "^1.1.1",
        "@reduxjs/toolkit": "^1.7.2",
        "@types/d3": "^6.2.0",
        "@types/geojson": "^7946.0.7",
        "@types/react": "^17.0.3",
        "@types/react-dom": "^17.0.3",
        "@types/react-redux": "^7.1.15",
        "@types/react-resize-detector": "^5.0.0",
        "@ungap/structured-clone": "^0.3.4",
        "@vivaxy/png": "^1.3.0",
        "@webviz/core-components": "^0.6.0",
        "ajv": "^7.2.1",
        "clsx": "^1.1.1",
        "convert-units": "^2.3.4",
        "d3": "^5.11.0",
        "d3-format": "^1.4.5",
        "deck.gl": "^8.8.11",
        "deep-equal": "^2.0.5",
        "fast-json-patch": "^3.0.0-1",
        "jsonschema": "^1.4.0",
        "jsverify": "^0.8.4",
        "leaflet": "^1.6.0",
        "leaflet-draw": "^1.0.4",
        "lodash": "^4.17.21",
        "mathjs": "^9.4.2",
        "nebula.gl": "^0.22.3",
        "react": "^17.0.2",
        "react-dom": "^17.0.2",
        "react-dropdown-tree-select": "^2.5.1",
        "react-leaflet": "2.8",
        "react-leaflet-draw": "^0.19.8",
        "react-redux": "^7.2.6",
        "react-resize-detector": "^7.0.0",
        "react-tooltip": "^4.2.21",
        "redux": "^4.0.5",
        "semver": "^7.3.5",
        "styled-components": "^5.3.3",
        "uuid": "^8.3.2"
    },
    "devDependencies": {
        "@actions/core": "^1.9.1",
        "@babel/core": "^7.13.10",
        "@babel/plugin-proposal-class-properties": "^7.13.0",
        "@babel/preset-env": "^7.13.12",
        "@babel/preset-react": "^7.12.13",
        "@babel/preset-typescript": "^7.13.0",
        "@babel/runtime": "^7.13.10",
        "@storybook/addon-docs": "^6.5.9",
        "@storybook/addon-essentials": "^6.5.9",
        "@storybook/addon-links": "^6.5.9",
        "@storybook/addon-storysource": "^6.5.9",
        "@storybook/builder-webpack5": "^6.5.9",
        "@storybook/manager-webpack5": "^6.5.9",
        "@storybook/react": "^6.5.9",
        "@testing-library/cypress": "^8.0.2",
        "@testing-library/jest-dom": "^5.11.9",
        "@testing-library/react": "^12.1.2",
        "@testing-library/user-event": "^13.1.9",
        "@types/cypress-image-snapshot": "^3.1.6",
        "@types/d3-color": "^3.0.2",
        "@types/d3-interpolate": "^3.0.1",
        "@types/jest": "^26.0.20",
        "@types/lodash": "^4.14.171",
        "@types/redux-mock-store": "^1.0.3",
        "@types/semver": "^7.3.8",
        "@types/ungap__structured-clone": "^0.3.0",
        "@types/uuid": "^8.3.0",
        "@typescript-eslint/eslint-plugin": "^4.19.0",
        "@typescript-eslint/parser": "^4.19.0",
        "ace-builds": "^1.4.12",
        "addon-redux": "^2.0.11",
        "babel-jest": "^26.6.3",
        "babel-loader": "^8.1.0",
        "concurrently": "^7.0.0",
        "copyfiles": "^2.0.0",
        "core-js": "^3.9.1",
        "css-loader": "^5.2.0",
        "css-minimizer-webpack-plugin": "^3.0.1",
        "cssnano": "^5.0.5",
        "cypress": "^9.4.1",
        "d3-color": "^3.1.0",
        "dotenv-webpack": "^7.0.2",
        "eslint": "^7.0.0",
        "eslint-config-prettier": "^8.1.0",
        "eslint-plugin-import": "^2.18.2",
        "eslint-plugin-prettier": "^3.1.3",
        "eslint-plugin-react": "^7.19.0",
        "file-loader": "^6.2.0",
        "hex-and-rgba": "^2.0.0",
        "html-webpack-plugin": "^5.3.1",
        "indefinitely-typed": "^1.1.0",
        "jest": "^26.6.3",
        "jest-styled-components": "^7.0.8",
        "json-schema": "^0.4.0",
        "mini-css-extract-plugin": "^1.3.9",
        "node-sass": "^7.0.3",
        "postcss": "^8.3.0",
        "prettier": "^2.5.1",
        "process": "^0.11.10",
        "raw-loader": "^4.0.2",
        "react-ace": "^9.5.0",
        "react-docgen": "^5.3.1",
        "react-test-renderer": "^17.0.2",
        "redux-mock-store": "^1.5.4",
        "sass": "^1.36.0",
        "sass-loader": "^12.1.0",
        "source-map-loader": "^3.0.0",
        "storybook-addon-performance": "^0.16.1",
        "terser-webpack-plugin": "^5.1.1",
        "trim": ">=0.0.3",
        "ts-loader": "^9.2.6",
        "typescript": "4.3.2",
        "url-loader": "^4.1.1",
        "vue-style-loader": "^4.1.3",
        "wait-on": "^6.0.1",
        "webpack": "^5.70.0",
        "webpack-cli": "^4.5.0",
        "webpack-conditional-loader": "^1.0.12",
        "webpack-dev-server": "^4.11.0"
    },
    "engines": {
        "node": ">=14.17",
        "npm": ">=6.14"
    }
}
