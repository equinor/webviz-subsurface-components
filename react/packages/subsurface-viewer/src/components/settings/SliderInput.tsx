import React, { FormEvent } from "react";
import { Label, Slider } from "@equinor/eds-core-react";

interface Props {
    /**
     * Label for the component.
     */
    label: string;
    /**
     * Initial state of the component.
     */
    value: number;
    /**
     * Min value.
     */
    min?: number;
    /**
     * Max value.
     */
    max?: number;
    /**
     * Stepping interval.
     */
    step?: number;
    /**
     * Callback to update the state of the component.
     */
    onChange: (e: FormEvent<HTMLDivElement>, value: number | number[]) => void;
}

const SliderInput: React.FC<Props> = React.memo(
    ({ label, value, min, max, step, onChange }: Props) => {
        return (
            <div
                style={{
                    display: "flex",
                    justifyContent: "space-between",
                }}
            >
                <Label
                    id={`${label}-slider-label`}
                    label={label}
                    style={{
                        paddingTop: 5,
                        fontSize: 15,
                    }}
                />
                <Slider
                    ariaLabelledby={`${label}-slider-label`}
                    id={`${label}-slider`}
                    value={value * 100}
                    min={min}
                    max={max}
                    step={step}
                    minMaxDots={false}
                    minMaxValues={false}
                    onChange={onChange}
                    style={{
                        paddingTop: 5,
                        paddingRight: 10,
                        paddingBottom: 25,
                        width: "3rem",
                    }}
                />
            </div>
        );
    }
);

SliderInput.displayName = "SliderInput";
export default SliderInput;
