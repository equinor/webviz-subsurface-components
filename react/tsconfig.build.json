{
    "extends": "./tsconfig.json",
    "exclude": ["**/storybook/", "**/test/", "**/*.test.tsx"]
}
