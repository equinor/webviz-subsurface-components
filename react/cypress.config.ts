import { defineConfig } from "cypress";
import getCompareSnapshotsPlugin from "cypress-image-diff-js/dist/plugin";

export default defineConfig({
  component: {
    devServer: {
      framework: "create-react-app",
      bundler: "webpack",
    },
    video:false,
    setupNodeEvents(on, config) {
      getCompareSnapshotsPlugin(on, config),
      on('before:browser:launch', (browser = {}, launchOptions) => {
        if (browser.family === 'chromium' && browser.name !== 'electron') {
          launchOptions.args.push('--start-fullscreen')
      
          return launchOptions
        }
        if (browser.name === 'electron') {
          launchOptions.preferences.fullscreen = true
      
          return launchOptions
        }
      })
    }
  },
});
