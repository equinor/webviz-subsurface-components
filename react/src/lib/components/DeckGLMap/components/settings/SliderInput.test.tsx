import { render } from "@testing-library/react";
import "jest-styled-components";
import "@testing-library/jest-dom";
import React, { FormEvent } from "react";
import { Wrapper } from "../../test/TestWrapper";
import SliderInput from "./SliderInput";

describe("Test Slider Input", () => {
    it("snapshot test", () => {
        const { container } = render(
            Wrapper({
                children: (
                    <SliderInput
                        label="Opacity"
                        value={50}
                        min={0}
                        max={100}
                        step={2}
                        onChange={(e: FormEvent<HTMLDivElement>) => {
                            e;
                        }}
                    />
                ),
            })
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
