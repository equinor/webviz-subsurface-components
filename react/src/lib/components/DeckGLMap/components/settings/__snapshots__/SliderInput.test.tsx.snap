// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Slider Input snapshot test 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
  margin: 0;
  color: var(--eds_text__static_icons__tertiary,rgba(111,111,111,1));
  font-family: Equinor;
  font-size: 0.750rem;
  font-weight: 500;
  line-height: 1.333em;
  text-align: left;
  margin-left: 8px;
  margin-right: 8px;
  color: var(--eds_text__static_icons__tertiary,rgba(111,111,111,1));
}

.c1 {
  margin: 0;
}

.c4 {
  --val: 5000;
  --realWidth: calc(100% - 12px);
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  position: relative;
  z-index: 1;
  margin: 0;
  color: var(--eds_text__static_icons__tertiary,rgba(111,111,111,1));
  font-family: Equinor;
  font-size: 0.625rem;
  font-weight: 500;
  line-height: 1.600em;
  -webkit-letter-spacing: 0.069em;
  -moz-letter-spacing: 0.069em;
  -ms-letter-spacing: 0.069em;
  letter-spacing: 0.069em;
  text-transform: uppercase;
  text-align: left;
  background: var(--eds_ui_background__default,rgba(255,255,255,1));
  padding: 0 5px;
  top: 24px;
  pointer-events: none;
  margin-left: calc((var(--val) - var(--min)) / var(--dif) * var(--realWidth));
  -webkit-transform: translate(calc(-1 * calc(var(--realWidth) / 2)));
  -ms-transform: translate(calc(-1 * calc(var(--realWidth) / 2)));
  transform: translate(calc(-1 * calc(var(--realWidth) / 2)));
  grid-row: 2;
  grid-column: 1 / -1;
}

.c3 {
  width: 100%;
  background: transparent;
  background: none;
  grid-column: 1 / -1;
  grid-row: 2;
  font: inherit;
  margin: 0;
  z-index: 2;
  outline: none;
}

.c3::-webkit-slider-runnable-track,
.c3::-webkit-slider-thumb,
.c3 {
  -webkit-appearance: none;
}

.c3::-moz-focus-outer {
  border: 0;
}

.c3[data-focus-visible-added]:focus {
  z-index: 2;
}

.c3[data-focus-visible-added]:focus::-webkit-slider-thumb {
  outline: 1px dashed var(--eds_interactive__focus,rgba(0,112,121,1));
  outline-offset: 2px;
}

.c3[data-focus-visible-added]:focus::-moz-range-thumb {
  outline: 1px dashed var(--eds_interactive__focus,rgba(0,112,121,1));
  outline-offset: 2px;
}

.c3:hover::-webkit-slider-thumb,
.c3:active::-webkit-slider-thumb {
  box-shadow: 0px 0px 0px 6px var(--eds_interactive_primary__hover_alt,rgba(222,237,238,1));
  border-color: var(--eds_interactive_primary__hover,rgba(0,79,85,1));
}

.c3:hover::-moz-range-thumb,
.c3:active::-moz-range-thumb {
  box-shadow: 0px 0px 0px 6px var(--eds_interactive_primary__hover_alt,rgba(222,237,238,1));
  border-color: var(--eds_interactive_primary__hover,rgba(0,79,85,1));
}

.c3:disabled::-webkit-slider-thumb {
  background-color: var(--eds_interactive__disabled__fill,rgba(234,234,234,1));
  border-color: var(--eds_ui_background__medium,rgba(220,220,220,1));
}

.c3:disabled::-moz-range-thumb {
  background-color: var(--eds_interactive__disabled__fill,rgba(234,234,234,1));
  border-color: var(--eds_ui_background__medium,rgba(220,220,220,1));
}

.c3:disabled:hover::-webkit-slider-thumb,
.c3:disabled:active::-webkit-slider-thumb {
  cursor: not-allowed;
  box-shadow: none;
}

.c3:disabled:hover::-moz-range-thumb,
.c3:disabled:active::-moz-range-thumb {
  cursor: not-allowed;
  box-shadow: none;
}

.c3:disabled:hover::-webkit-slider-runnable-track,
.c3:disabled:active::-webkit-slider-runnable-track {
  cursor: not-allowed;
}

.c3:disabled:hover::-moz-range-track,
.c3:disabled:active::-moz-range-track {
  cursor: not-allowed;
}

.c3:after {
  right: 0;
}

.c3::-webkit-slider-thumb {
  border: 2px solid var(--eds_interactive_primary__resting,rgba(0,112,121,1));
  border-radius: 50%;
  height: 12px;
  width: 12px;
  background: var(--eds_ui_background__default,rgba(255,255,255,1));
  cursor: pointer;
  position: relative;
  z-index: 1;
  margin-top: 6px;
}

.c3::-moz-range-thumb {
  border: 2px solid var(--eds_interactive_primary__resting,rgba(0,112,121,1));
  border-radius: 50%;
  height: 12px;
  width: 12px;
  background: var(--eds_ui_background__default,rgba(255,255,255,1));
  cursor: pointer;
  position: relative;
  z-index: 1;
  height: 8px;
  width: 8px;
}

.c3::-webkit-slider-runnable-track {
  width: 100%;
  height: 100%;
  cursor: pointer;
  background: none;
}

.c3::-moz-range-track {
  width: 100%;
  height: 100%;
  cursor: pointer;
  background: none;
}

.c2 {
  --min: 0;
  --max: 100;
  --dif: calc(var(--max) - var(--min));
  --value: 5000;
  --realWidth: calc(100% - 12px);
  display: grid;
  grid-template-rows: max-content 24px;
  grid-template-columns: 1fr 1fr;
  width: 100%;
  position: relative;
  background-image: url("data:image/svg+xml,<svg xmlns='http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg'><rect x='0' y='11' fill='var(--eds_ui_background__light, rgba(247, 247, 247, 1))' width='100%' height='4' rx='2' /></svg>");background-size:cover;background-repeat:no-repeat;} .c2::after{grid-column:1 / span 2;grid-row:2;height:4px;margin-bottom:9px;-webkit-align-self:end;-ms-flex-item-align:end;align-self:end;content:'';background:var(--eds_interactive_primary__resting,rgba(0,112,121,1));} .c2::after{margin-right:calc( (var(--max) - var(--value)) / var(--dif) * var(--realWidth) );margin-left:3px;} .c2:hover:not([disabled]){background-image:url("data:image/svg+xml,<svg xmlns='http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg'><rect x='0' y='11' fill='var(--eds_ui_background__medium, rgba(220, 220, 220, 1))' width='100%' height='4' rx='2' /></svg>");
}

.c2:hover:not([disabled])::after {
  background: var(--eds_interactive_primary__hover,rgba(0,79,85,1));
}

<div
  style="display: flex; justify-content: space-between;"
>
  <label
    class="c0"
    id="Opacity-slider-label"
    style="padding-top: 5px; font-size: 15px;"
  >
    <span
      class="c1"
    >
      Opacity
    </span>
  </label>
  <div
    class="c2"
    id="Opacity-slider"
    max="100"
    min="0"
    style="padding-top: 5px; padding-right: 10px; padding-bottom: 25px; width: 3rem;"
    value="5000"
  >
    <input
      aria-labelledby="Opacity-slider-label"
      class="c3"
      id="Opacity-slider-label-thumb"
      max="100"
      min="0"
      step="2"
      type="range"
      value="5000"
    />
    <output
      class="c4"
      for="Opacity-slider-label-thumb"
      value="5000"
    >
      5000
    </output>
  </div>
</div>
`;
